### ISSUE TITLE
KAS-IFC "VAL" Potential Bug

### ISSUE THREAD

**Issue by powersmc** [usnistgov/ACVP-Server#97]
**environment**
Prod

**testSessionId**
5966

**vsId**
58398

**Algorithm registration**
[
  {
    "acvVersion":"1.0"
  },
  {
    "algorithms":[
      {
        "algorithm":"KAS-IFC",
        "revision":"Sp800-56Br2",
        "function":[
          "keyPairGen",
          "partialVal"
        ],
        "iutId":"48534d",
        "keyGenerationMethods":[
          "rsakpg1-crt",
          "rsakpg2-crt"
        ],
        "modulo":[
          4096
        ],
        "fixedPubExp":"010001",
        "scheme":{
          "KAS1-basic":{
            "kasRole":[
              "initiator",
              "responder"
            ],
            "kdfMethods":{
              "oneStepKdf":{
                "auxFunctions":[
                  {
                    "auxFunctionName":"SHA2-512"
                  }
                ],
                "fixedInfoPattern":"uPartyInfo||vPartyInfo",
                "encoding":[
                  "concatenation"
                ]
              }
            },
            "l":1024
          }
        }
      },
      {
        "algorithm":"KTS-IFC",
        "revision":"Sp800-56Br2",
        "function":[
          "keyPairGen",
          "partialVal"
        ],
        "iutId":"48534d",
        "keyGenerationMethods":[
          "rsakpg1-crt",
          "rsakpg2-crt"
        ],
        "modulo":[
          2048,
          3072,
          4096,
          6144,
          8192
        ],
        "fixedPubExp":"010001",
        "scheme":{
          "KTS-OAEP-basic":{
            "kasRole":[
              "initiator",
              "responder"
            ],
            "ktsMethod":{
              "hashAlgs":[
                "SHA2-224",
                "SHA2-256",
                "SHA2-384",
                "SHA2-512",
                "SHA3-224",
                "SHA3-256",
                "SHA3-384",
                "SHA3-512"
              ],
              "supportsNullAssociatedData":true,
              "associatedDataPattern":"uPartyInfo||vPartyInfo",
              "encoding":[
                "concatenation"
              ]
            },
            "l":256
          }
        }
      }
    ]
  }
]

**Endpoint in which the error is experienced**
https://acvts.nist.gov/acvp/v1/testSessions/5966/vectorSets/58398

**Expected behavior**
Based on the specification here:
https://pages.nist.gov/ACVP/draft-hammett-acvp-kas-ifc.html#name-test-types

It states:
"Various types of errors **MUST** be introduced in varying portions of the key agreement process (changed DKM, changed key, changed hash digest, etc.), that the IUT **MUST** be able to detect and report on.”

So the expectation would be that in the "VAL" tests you would expect to see at least one error injected.  In this particular case, every single "VAL" test was marked as "**testPassed": true** in the results and the server marked them all as passed (i.e.: for tgId 2, tcId 11-35)

**Additional context**
This could just be an error in the spec, where instead of the first "MUST" it might be more appropriate to put "MAY".  Either that, or the spec is right and there's a server-side issue where it's not guaranteeing at least one error every time as it should.

**Comment by Kritner** [comment]
@powersmc have you seen any additional cases of this occurring regarding no expected failure tests in kas-ifc? We have had a few recent changes to some vector set generation to kas-ifc, so I'm wondering if it could have been fixed in one of those changes.  I just tried running the same registration you provided:

```json
{
	"algorithm": "KAS-IFC",
	"revision": "Sp800-56Br2",
	"function": [
		"keyPairGen",
		"partialVal"
	],
	"iutId": "48534d",
	"keyGenerationMethods": [
		"rsakpg1-crt",
		"rsakpg2-crt"
	],
	"modulo": [
		4096
	],
	"fixedPubExp": "010001",
	"scheme": {
		"KAS1-basic": {
			"kasRole": [
				"initiator",
				"responder"
			],
			"kdfMethods": {
				"oneStepKdf": {
					"auxFunctions": [{
						"auxFunctionName": "SHA2-512"
					}],
					"fixedInfoPattern": "uPartyInfo||vPartyInfo",
					"encoding": [
						"concatenation"
					]
				}
			},
			"l": 1024
		}
	}
}
```

and did get some expected failures within the val group.  The spec is correct here, you *should* be getting at a minimum one failure scenario per val group; I can confirm the mentioned vs 58398 did not have any however.

I'm pretty sure there was an issue closed about this within the past month or two...

**Comment by powersmc** [comment]
@Kritner No, I haven't seen the issue since then.  That being said, the vector set 58398 was generated on production on 03/29/2021 from what I'm seeing, so based on [this page](https://github.com/usnistgov/ACVP-Server/releases) the only release that has come out since then on production was v1.1.0.17, and I'm not seeing anything about KAS-IFC in those release notes.

**Comment by Kritner** [comment]
This change has been released to prod in release [v1.1.0.18](https://github.com/usnistgov/ACVP-Server/releases/tag/v1.1.0.18).
